component/prometheus/scrape/scrape.go

```diff
@@ -236,13 +228,22 @@ func (c *Component) Update(args component.Arguments) error {
 	defer c.mut.Unlock()
 	c.args = newArgs
 
+	// Update scraper with scrape options
+	data, err := c.opts.GetServiceData(http.ServiceName)
+	if err != nil {
+		return fmt.Errorf("failed to get information about HTTP server: %w", err)
+	}
+	newFlowAppendables, newScraper := c.createPromScrapeResources(data.(http.Data), newArgs)
+	c.appendable = newFlowAppendables
+	c.scraper = newScraper
```

Comment:
There's an issue here: the Run method runs the original `c.scraper`, and will never terminate that scraper/run the new one that's created. This means the changes here on subsequent updates will never truly apply.

Reply:
I totally missed this one out 🫣 

But I do think we can rewrite the scraper's Run goroutine into a for-select loop to handle scraper restart. 

Let me try out my luck on this.